server/src/db/catalog/chunk.rs

```diff
@@ -610,30 +617,35 @@ impl CatalogChunk {
     /// Set the chunk in the Moved state, setting the underlying
     /// storage handle to db, and discarding the underlying mutable buffer
     /// storage.
-    pub fn set_moved(&mut self, chunk: Arc<RBChunk>) -> Result<()> {
+    pub fn set_moved(&mut self, chunk: Arc<RBChunk>, schema: Schema) -> Result<()> {
         match &mut self.stage {
-            ChunkStage::Frozen { representation, .. } => match &representation {
-                ChunkStageFrozenRepr::MutableBufferSnapshot(_) => {
-                    self.metrics
-                        .state
-                        .inc_with_labels(&[KeyValue::new("state", "moved")]);
+            ChunkStage::Frozen {meta,  representation, .. } => {
+                // after moved, the chunk is sorted and schema need to get updated
+                let mut metadata = meta.lock();
```

Comment:
If this is the only use of the mutex, I think you can avoid using it with something like:

```suggestion
            ChunkStage::Frozen {mut metadata,  representation, .. } => {
                // after moved, the chunk is sorted and schema need to get updated
```